rotkehlchen/chain/arbitrum_one/modules/umami/utils.py

```diff
@@ -0,0 +1,54 @@
+
+import logging
+from typing import TYPE_CHECKING
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.arbitrum_one.modules.umami.constants import UMAMI_ASSET_VAULT_ABI
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import Price
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.inquirer import Inquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def get_umami_vault_token_price(
+        inquirer: 'Inquirer',
+        vault_token: 'EvmToken',
+        evm_inquirer: 'EvmNodeInquirer',
+) -> Price | None:
+    """Gets token price for Umami vault.
+
+    Multiplies vault's price per share by the underlying token's USD price.
+
+    Args:
+        inquirer: For underlying token price lookup.
+        vault_token: Vault token to get price data for.
+        evm_inquirer: Inquirer instance for making contract calls.
+
+    Returns:
+        Price: USD price of underlying token, None if pricing fails
+    """
+    try:
+        price_per_share = evm_inquirer.call_contract(
+            contract_address=vault_token.evm_address,
+            abi=UMAMI_ASSET_VAULT_ABI,
+            method_name='pps',
+        )
+    except RemoteError as e:
+        log.error(f'Failed to get price per share for vault {vault_token.evm_address}: {e}')
+        return None
```

Comment:
why not return zero here too? In the places you call this you end up having to do an extra check for `None` and one for `ZERO`.

you can log the error and return zero price.